publisher/publisher.go

```diff
@@ -245,8 +245,18 @@ func (pub *Impl) SubmitToSingleCTWithResult(ctx context.Context, req *pubpb.Requ
 		ctLog)
 	if err != nil {
 		if !canceled.Is(err) {
-			pub.log.AuditErr(
-				fmt.Sprintf("Failed to submit certificate to CT log at %s: %s", ctLog.uri, err))
+			if respErr, ok := err.(ctClient.RspError); ok && respErr.StatusCode < 500 {
```

Comment:
This `if` nesting is getting kinda un-Go-like. How about just one level of nesting:

```
if err != nil {
  if canceled.Is(err) {
    return nil, err
  }
  var body string
  if respErr, ok := ... { // type assertions on nil with ", ok" work okay
    body = "..."
  }
  pub.log.AuditErr(...)
  return nil, err
}
```